pubtools/_pulp/tasks/copy_repo.py

```diff
@@ -0,0 +1,324 @@
+import logging
+import sys
+from functools import partial
+
+import attr
+from more_executors.futures import f_flat_map, f_map, f_sequence
+from pubtools.pulplib import (
+    ContainerImageRepository,
+    Criteria,
+    FileUnit,
+    ModulemdUnit,
+    PublishOptions,
+    RpmUnit,
+)
+from pushsource import FilePushItem, ModuleMdPushItem, RpmPushItem
+
+from pubtools._pulp.arguments import SplitAndExtend
+from pubtools._pulp.services import (
+    CollectorService,
+    PulpClientService,
+    UdCacheClientService,
+)
+from pubtools._pulp.task import PulpTask
+from pubtools._pulp.tasks.common import CDNCache
+
+step = PulpTask.step
+
+LOG = logging.getLogger("pubtools.pulp")
+
+
+@attr.s(slots=True)
+class RepoCopy(object):
+    """Represents a copy of a single repo."""
+
+    tasks = attr.ib()
+    """The completed Pulp tasks for copying."""
+
+    repo = attr.ib()
+    """The repo to which content was copied."""
+
+
+class CopyRepo(
+    CollectorService, UdCacheClientService, PulpClientService, CDNCache, PulpTask
+):
+    @property
+    def content_type(self):
+        # Only return non-None if there were really any types given.
+        # Otherwise, return None to let library defaults apply
+        return self.args.content_type or None
+
+    @property
+    def repo_pairs(self):
+        return [pair.split(",") for pair in self.args.repopairs]
+
+    def add_args(self):
+        super(CopyRepo, self).add_args()
+
+        self.parser.add_argument(
+            "--skip", help="skip given comma-separated sub-steps", type=str
+        )
+        self.parser.add_argument(
+            "--content-type",
+            help="copy only content of these comma-separated type(s). e.g. --content-type=(rpm, srpm, modulemd, iso)",
+            type=str,
+            action=SplitAndExtend,
+            split_on=",",
+        )
+        self.parser.add_argument(
+            "repopairs",
+            type=str,
+            nargs="+",
+            help="repository pair(s) (src, dest) to be copied. e.g. repo-A,repo-B repo-C,repo-D",
+        )
+
+    def fail(self, *args, **kwargs):
+        LOG.error(*args, **kwargs)
+        sys.exit(30)
+
+    @step("Check repos")
+    def get_repos(self, repos):
+        # Returns all pairs of repos to be operated on by this task.
```

Comment:
This comment doesn't seem accurate.  It looks like `repos` is expected to be a single pair only.  If this were not the case, the code a bit later on which looks like `if repo.id == repos[0]:` would be wrong as there is never any attempt to process anything beyond the first two elements of `repos`.

If it's designed to work on only a single pair at a time, may I suggest that `get_repos(self, src_repo, dst_repo):` or similar would be clearer?